### PR TITLE
Remove nw-creating-secured-routes

### PR DESCRIPTION
**Pull request by Miciah**
#### Remove `nw-creating-secured-routes`

Delete a redundant module that describes how to create an edge route.  Another module already describes the procedure.  The latter module does not describe how to create a self-signed certificate, but doing so is out of scope for the product documentation.

* `modules/nw-creating-secured-routes.adoc`: Delete file.
* `modules/nw-ingress-creating-an-edge-route-with-a-custom-certificate.adoc`: Add a couple sentences from the deleted module.
* `networking/ingress/route-configuration.adoc`: Delete the include stanza for the deleted module.

#### Rearrange ingress topics

* `_topic_map.yml`: Delete "in OpenShift Container Platform" from the "Understanding the DNS Operator" topic title.  Group topics related to the ingress operator under the "Understanding the Ingress Operator" topic. Group topics related to the Route resource under the "Configuring Routes" topic.  Put both topics directly under "Networking".
* `networking/ingress/configuring-default-certificate.adoc`: Delete file. This assembly's modules are absorbed into the `understanding-the-ingress-operator` and `secured-routes` assemblies.
* `networking/ingress/route-configuration.adoc`: Moved...
* `networking/routes/route-configuration.adoc`: ...here.
* `networking/routes/route-types/secured-routes.adoc`: New assembly.  Include the `nw-ingress-creating-a-reencrypt-route-with-a-custom-certificate` and `nw-ingress-creating-an-edge-route-with-a-custom-certificate` modules.
* `networking/ingress/ingress-controller-configuration.adoc`: Delete file. This assembly's module is absorbed into the `understanding-the-ingress-operator` assembly.
* `networking/ingress/configuring-ingress-operator.adoc`: Rename...
* `networking/ingress-operator.adoc`: ...to this. Include the `nw-ingress-setting-a-custom-default-certificate` and `nw-scaling-ingress-controller` assemblies.

---

If anyone objects to my assertion that creating a self-signed certificate does not belong in the product documentation, then I propose putting it in a separate module that can be included before the `nw-ingress-creating-a-reencrypt-route-with-a-custom-certificate` and `nw-ingress-creating-an-edge-route-with-a-custom-certificate` modules.  Let me know, and I'll amend this PR.

### CODE REVIEW

**Comment by bmcelvee** [comment]
@lihongan PTAL, thanks!

**Comment by lihongan** [comment]
I think this is belong to `Routes`, so `nw-creating-secured-routes` and `nw-ingress-creating-an-edge-route-with-a-custom-certificate` should be moved from `Ingress Operator` to `Routes`.

My proposal is to creat two sections under `Networking`, like:
>Networking
-->Understanding Ingress Operator
-->Configuring Routes
     -->Route types
         -->Secured routes (contains edge, pass through, re-encrypt termination)

I've opened a bug to optimize routes topic: https://bugzilla.redhat.com/show_bug.cgi?id=1707750

**Comment by Miciah** [comment]
> I think this is belong to `Routes`, so `nw-creating-secured-routes` and `nw-ingress-creating-an-edge-route-with-a-custom-certificate` should be moved from `Ingress Operator` to `Routes`.

Do you agree that `nw-creating-secured-routes` and `nw-ingress-creating-an-edge-route-with-a-custom-certificate` are redundant, and that we can delete one of them?

> My proposal is to creat two sections under `Networking`, like:
> 
> > Networking
> > -->Understanding Ingress Operator

"Understanding Ingress Operator" the existing "Ingress Operator in OpenShift Container Platform" content, or is it yet-to-be-written content?

> > -->Configuring Routes
> > ------>Route types
> > --------->Secured routes (contains edge, pass through, re-encrypt termination)

Done.  Let me know whether I interpreted the suggestion correctly.  I put ingress operator and ingress controller content all under "Understanding the Ingress Operator"—is that all right? It makes sense to me, but I want to make sure it is what you had in mind.

File: _topic_map.yml [hunk 1]
```diff
@@ -366,14 +366,18 @@ Topics:
 - Name: Ingress
   Dir: ingress
```
Comment (lihongan) [inline _topic_map.yml]: I'd like to change above two lines to below:
```
 - Name: Understanding the Ingress Operator
   File: ingress-operator
 - Name: Configuring Routes
   File: configuring-routes
```
This is what I really want, I think "DNS operator", "Ingress Operator" and "Routes" are same level.

File: _topic_map.yml [hunk 2]
```diff
@@ -366,14 +366,18 @@ Topics:
...
   Topics:
-  - Name: Ingress Operator in OpenShift Container Platform
-    File: configuring-ingress-operator
-  - Name: Configuring the default certificate
-    File: configuring-default-certificate
-  - Name: Ingress controller configuration
-    File: ingress-controller-configuration
```
Comment (lihongan) [inline _topic_map.yml]: Above three topics should be in `File: ingress-operator`.

**Comment by lihongan** [comment]
> > I think this is belong to `Routes`, so `nw-creating-secured-routes` and `nw-ingress-creating-an-edge-route-with-a-custom-certificate` should be moved from `Ingress Operator` to `Routes`.
> 
> Do you agree that `nw-creating-secured-routes` and `nw-ingress-creating-an-edge-route-with-a-custom-certificate` are redundant, and that we can delete one of them?
> 
Agree 
> > My proposal is to creat two sections under `Networking`, like:
> > > Networking
> > > -->Understanding Ingress Operator
> 
> Is "Understanding Ingress Operator" the existing "Ingress Operator in OpenShift Container Platform" content, or is it yet-to-be-written content?
> 
Here `Understanding Ingress Operator` is the higher level title, I've added comments to the part.
> > > -->Configuring Routes
> > > ------>Route types
> > > --------->Secured routes (contains edge, pass through, re-encrypt termination)
> 
> Done. Let me know whether I interpreted the suggestion correctly. I put ingress operator and ingress controller content all under "Understanding the Ingress Operator"—is that all right? It makes sense to me, but I want to make sure it is what you had in mind.
`Configuring Routes` is also the higher level title, I think its contents source from 3.11 and just need minor updates.

File: _topic_map.yml (hunk 1 above)
Comment (Miciah) [reply _topic_map.yml]: Done.

File: _topic_map.yml (hunk 2 above)
Comment (Miciah) [reply _topic_map.yml]: I made that change in the previous revision to this PR: the `ingress-operator` assembly includes the `nw-ingress-setting-a-custom-default-certificate` as well as the `nw-scaling-ingress-controller` module that was in the `ingress-controller-configuration` assembly.  Is that what you mean, or should I make some further changes?

**Comment by lihongan** [comment]
LGTM - thank you @Miciah

**Comment by Miciah** [comment]
Rebased to fix a conflict from https://github.com/openshift/openshift-docs/pull/14728.

https://github.com/openshift/openshift-docs/pull/14728 moved the "DNS Operator" topic underneath a new "DNS" directory, whereas this PR flattens the "Ingress" directory into the "Networking" directory, so I wonder whether this PR should also revert the change that adds the new "DNS" directory.

**Comment by jboxman** [comment]
@bmcelvee, I've been pondering our _Networking_ TOC and wondering how best to structure it.

**Comment by bmcelvee** [comment]
> @bmcelvee, I've been pondering our _Networking_ TOC and wondering how best to structure it.

@jboxman that's fair. Once we get more content in place, we should certainly take a look to see how we can optimize the structure.

**Comment by Miciah** [comment]
Rebased.

**Comment by Miciah** [comment]
Rebased.

**Comment by Miciah** [comment]
Rebased.

**Comment by vikram-redhat** [comment]
@bmcelvee @jboxman were we going ahead with these changes for 4.1?

**Comment by jboxman** [comment]
@vikram-redhat, good question.

**Comment by jboxman** [comment]
But definitely not as is -- the build fails.

**Comment by jboxman** [comment]
Two different issues with `_topic_map.yml`:

1. Invalid syntax
1. Nested > 2 levels (revealed after 1 fixed)

So we need to flatten this out by 1 level.

**Comment by jboxman** [comment]
I opened https://github.com/openshift/openshift-docs/pull/15091 which fixes the build errors.

**Comment by vikram-redhat** [comment]
Closing this.

**Pull request closed without merging**